File: InventoryDemo/app/src/main/java/com/psl/inventorydemo/adapter/SearchableAdapter.java
package com.psl.inventorydemo.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.Filter;
import android.widget.Filterable;
import android.widget.TextView;

import com.psl.inventorydemo.R;

import java.util.ArrayList;
import java.util.List;

public class SearchableAdapter extends BaseAdapter implements Filterable {

    private Context context;
    private List<String> originalData;
    private List<String> filteredData;

    public SearchableAdapter(Context context, List<String> data) {
        this.context = context;
        this.originalData = data;
        this.filteredData = data;
    }

    @Override
    public int getCount() {
        return filteredData.size();
    }

    @Override
    public Object getItem(int position) {
        return filteredData.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            LayoutInflater inflater = LayoutInflater.from(context);
            convertView = inflater.inflate(android.R.layout.simple_list_item_1, parent, false);
            if (position % 2 != 0) {
                convertView.setBackgroundColor(context.getResources().getColor(R.color.cyan1));
            } else {
                convertView.setBackgroundColor(context.getResources().getColor(R.color.lemonyellow));
            }
        }

        String item = filteredData.get(position);

        // Customize the view as needed
        TextView textView = convertView.findViewById(android.R.id.text1);
        textView.setText(item);

        return convertView;
    }

    @Override
    public Filter getFilter() {
        return new Filter() {
            @Override
            protected FilterResults performFiltering(CharSequence charSequence) {
                String filterString = charSequence.toString().toLowerCase();

                FilterResults results = new FilterResults();

                final List<String> list = originalData;

                int count = list.size();
                final ArrayList<String> nlist = new ArrayList<>(count);

                for (String item : list) {
                    if (item.toLowerCase().contains(filterString)) {
                        nlist.add(item);
                    }
                }

                results.values = nlist;
                results.count = nlist.size();

                return results;
            }

            @Override
            protected void publishResults(CharSequence charSequence, FilterResults filterResults) {
                filteredData = (ArrayList<String>) filterResults.values;
                notifyDataSetChanged();
            }
        };
    }
}

File: app/src/main/java/com/example/myexcel/SearchableAdapter.kt
package com.example.myexcel

import android.R
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.Filter
import android.widget.Filterable
import android.widget.TextView
import java.util.Locale


class SearchableAdapter(private val context: Context, private val originalData: ArrayList<String>) :
    BaseAdapter(), Filterable {
    private var filteredData: ArrayList<String>

    init {
        this.filteredData = originalData
    }

    override fun getCount(): Int {
        return filteredData.size
    }

    override fun getItem(position: Int): Any {
        return filteredData[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        var convertView = convertView
        if (convertView == null) {
            val inflater = LayoutInflater.from(context)
            convertView = inflater.inflate(R.layout.simple_list_item_1, parent, false)

        }

        val item = filteredData[position]

        // Customize the view as needed
        val textView = convertView?.findViewById<TextView>(R.id.text1)
        if (textView != null) {
            textView.text = item
        }

        return convertView
    }

    override fun getFilter(): Filter {
        return object : Filter() {
            override fun performFiltering(charSequence: CharSequence): FilterResults {
                val filterString = charSequence.toString().lowercase(Locale.getDefault())

                val results: FilterResults = FilterResults()

                val list = originalData

                val count = list.size
                val nlist = ArrayList<String>(count)

                for (item in list) {
                    if (item.lowercase(Locale.getDefault()).contains(filterString)) {
                        nlist.add(item)
                    }
                }

                results.values = nlist
                results.count = nlist.size

                return results
            }

            override fun publishResults(
                charSequence: CharSequence?,
                filterResults: FilterResults
            ) {
                filteredData = filterResults.values as ArrayList<String>
                notifyDataSetChanged()
            }
        }
    }
}